doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,394 @@
+>>> solution = solve(x ** 4 - 256, x)
+>>> print(solution)
+[-4, 4]
+```
+
+### Substituting solutions from {func}`~.solve` into an expression
+
+You can substitute solutions from {func}`~.solve` into an expression.
+
+A common use case is finding the critical points and values for a function $f$. 
+At the critical points, the derivative equals zero (or is undefined). 
+You can then obtain the function values at those critical points
+by substituting the critical points back into the function using {meth}`~sympy.core.basic.Basic.subs`.
+
+```py
+>>> from sympy.abc import x
+>>> from sympy import solve, diff
+>>> f = x**3 + x**2 - x
+>>> derivative = diff(f, x)
+>>> critical_points = solve(derivative, x, dict=True)
+>>> print(critical_points)
+[{x: -1}, {x: 1/3}]
+>>> for critical_point in critical_points:
+...     # Extract the value from the dictionary where key is x
+...     critical_point_value = critical_point[x]
+...     f_at_critical_point = f.subs(x, critical_point_value)
+...     print(f"f({critical_point_value}) = {f_at_critical_point}")
+f(-1) = 1
+f(1/3) = -5/27
+```
+
+## Using {func}`~.solveset`
+
+- produces outputs in the format of 
+[SymPy mathematical Sets](https://docs.sympy.org/dev/modules/sets.html?highlight=sets#module-sympy.sets.sets) rather than 
+[Python sets](https://docs.python.org/3/library/stdtypes.html#set)
+- can return infinitely many solutions
+- the solution set can be more difficult to parse programmatically
+
+You can solve an equation using solve() in several ways.
+
+### Make your equation into an expression that equals zero
+
+Use the fact that any expression not in an `Eq` (equation) is automatically assumed to equal zero (0)
+ by the solving functions. You can rearrange the equation $x^2 = y$ to $x^2 - y = 0$, and 
+ {func}`~.solveset` that expression. This approach is convenient if you are interactively solving an 
+expression which already equals zero, or an equation that you do not mind rearranging to $expression = 0$.
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution = solveset(x**2 - y, x)
+>>> print(solution)
+{-sqrt(y), sqrt(y)}
+```
+
+### Put your equation into `Eq` form
+
+Put your equation into `Eq` form, then apply {func}`~.solveset` to the `Eq`. 
+This approach is convenient if you are interactively solving an equation which you already have in the form of an equation,
+or which you think of as an equality.
+
+```py
+>>> from sympy import Eq, solveset
+>>> from sympy.abc import x, y
+>>> eqn = Eq(x**2, y)
+>>> solution = solveset(eqn, x)
+>>> print(solution)
+{-sqrt(y), sqrt(y)}
+```
+
+### Parse a string representing the equation
+Parse a string representing the equation into a form that SymPy can understand (`Eq` form), then apply 
+{func}`~.solveset` to the parsed expression.  This approach is convenient if you are programmatically 
+reading in a string. We [recommend against using parsing a string if you are creating the expression 
```

Comment:
I don't think this needs to be repeated.

Reply:
Same for "eq form" above.

Reply:
By "this" do you mean just the narrative preambles, or also the code examples?

Reply:
The whole section. There is a section above about parsing expressions and using that with `solve`. This section is basically a repetition of that. Same goes for "put your equation in Eq form". We don't need to duplicate these but rather can note once that an equation parsed from a string can be passed to either solve or solveset and both accept Eq or Expr.

Reply:
I had considered reorganizing the guide to be by solving approach (expression, `Eq`, or parsing) first and then by type (`solve` or `solveset`). I did that to address your feedback, make the guide shorter, and let visitors easily compare the output of `solve` or `solveset`.